### PR TITLE
Update validation for github owner to allow _

### PR DESCRIPTION
**Pull request by shahzad31**
Update validation for github owner to allow _ !!

follow up to https://github.com/elastic/elastic-package/pull/1564

### CODE REVIEW

**Review by jsoriano — COMMENTED** [review]
Could you please add a changelog entry in `spec/changelog.yml`, and an example of usage in a test package? Thanks!

**Comment by jsoriano** [comment]
Oh, and looking to the failures in CI you will need to update the pattern that appears in error messages in tests.

File: code/go/pkg/validator/validator_test.go [hunk 1]
```diff
@@ -78,10 +78,10 @@ func TestValidateFile(t *testing.T) {
 				"field policy_templates.2.data_streams.1: data stream doesn't exist",
 			},
 		},
-		"bad_github_owner": {
+		"bad_github_owner/": {
```
Comment (jsoriano) [inline code/go/pkg/validator/validator_test.go:81]: @shahzad31 is this change required?

File: spec/changelog.yml [hunk 2]
```diff
@@ -636,3 +636,6 @@
   - description: Introduce "value" property for constant_keyword fields
     type: enhancement
     link: https://github.com/elastic/package-spec/pull/194
+  - description: Allow _ in github owner field
+    type: enhancement
+    link: https://github.com/elastic/package-spec/pull/668
```
Comment (jsoriano) [inline spec/changelog.yml:641]: This change should be moved to the section for the next version, top of the file.

File: spec/integration/manifest.spec.yml [hunk 3]
```diff
@@ -251,7 +251,11 @@ spec:
         github:
           description: Github team name of the package maintainer.
           type: string
-          pattern: '^(([a-zA-Z0-9-]+)|([a-zA-Z0-9-]+\/[a-zA-Z0-9-]+))$'
+          pattern: '^(([a-zA-Z0-9-_]+)|([a-zA-Z0-9-_]+\/[a-zA-Z0-9-_]+))$'
+          examples:
+                - "elastic"
+                - "apm-agent-java"
+                - "ux_infra_team"
```
Comment (jsoriano) [inline spec/integration/manifest.spec.yml:258]: :+1: thanks for adding examples too.

**Review by jsoriano — CHANGES_REQUESTED** [review]

File: code/go/pkg/validator/validator_test.go (hunk 1 above)
Comment (shahzad31) [reply code/go/pkg/validator/validator_test.go:81]: bad_github_owner wasn't bad anymore :) 

/ makes it bad now :)

Comment (jsoriano) [reply code/go/pkg/validator/validator_test.go:81]: This is the name of the directory with the package, it should not make any difference to have the slash or not.
I would prefer not having it for consistency with the rest of packages listed here.

File: spec/changelog.yml (hunk 2 above)
Comment (jsoriano) [reply spec/changelog.yml]: Fixed.

**Review by jsoriano — APPROVED** [review]
